tests/test_unit.py

```diff
@@ -0,0 +1,375 @@
+        f = open(filename, "w")
+        f.write("""
+# generated by jobsub_lite 
+# 
+universe           = vanilla
+executable         = /bin/true
+arguments          = 
+
+output             = lookaround.xx.$(Cluster).$(Process).out
+error              = lookaround.xx.$(Cluster).$(Process).err
+log                = lookaround.xx.$(Cluster).$(Process).log
+environment        = CLUSTER=$(Cluster);PROCESS=$(Process);CONDOR_TMP=;BEARER_TOKEN_FILE=.condor_creds/{group}.use;CONDOR_EXEC=/tmp;DAGMANJOBID=$(DAGManJobId);GRID_USER={user};JOBSUBJOBID=$(CLUSTER).$(PROCESS)@jobsubdevgpvm01.fnal.gov;EXPERIMENT={group};SAM_EXPERIMENT=samdev
+rank               = Mips / 2 + Memory
+job_lease_duration = 3600
+notification       = Never
+transfer_output    = True
+transfer_error     = True
+transfer_executable= True
+when_to_transfer_output = ON_EXIT_OR_EVICT
+transfer_output_files = .empty_file
+request_memory = 2048.0
+request_disk = 102400.0KB
++JobsubClientDN=""
++JobsubClientIpAddress="131.225.67.71"
++Owner="{user}"
++JobsubServerVersion="lite_v1_0"
++JobsubClientVersion="lite_v1_0"
++JobsubClientKerberosPrincipal=""
++JOB_EXPECTED_MAX_LIFETIME = 28800.0
+notify_user = {user}@fnal.gov
++AccountingGroup = "group_{group}.{user}"
++Jobsub_Group="{group}"
++JobsubJobId="$(CLUSTER).$(PROCESS)@jobsubdevgpvm01.fnal.gov"
++Drain = False
++GeneratedBy =" jobsubdevgpvm01.fnal.gov"
+
++DESIRED_usage_model="OPPORTUNISTIC,DEDICATED"
+
+requirements  = target.machine =!= MachineAttrMachine1 && target.machine =!= MachineAttrMachine2  && (isUndefined(DesiredOS) || stringListsIntersect(toUpper(DesiredOS),IFOS_installed)) && (stringListsIntersect(toUpper(target.HAS_usage_model, toUpper(my.DESIRED_usage_model))))
+
+#
+# this is supposed to get us output even if jobs are held(?)
+#
++SpoolOnEvict = false
+#
+#
+#
+use_oauth_services = {group}
+
+queue 1
+        """.format( user = os.environ["USER"], group = TestUnit.test_group ))
+        f.close()
+        return filename
+
+    def get_dag_file():
+        filename="/tmp/tst{0}.dag".format(os.getpid())
+        f = open(filename, "w")
+        f.write("""
+        """)
+        f.close()
+        return filename
+
+    # lib/get_parser.py routines...
+
+    def test_get_parser_1(self):
+        parser = get_parser.get_parser()
+        line = "jobsub_submit --devserver -e SAM_EXPERIMENT -G {0} --resource-provides=usage_model=OPPORTUNISTIC,DEDICATED,OFFSITE file://`pwd`/lookaround.sh".format(TestUnit.test_group)
+        res = parser.parse_args(line.strip().split()[1:])
+        assert res.devserver
+        assert 'SAM_EXPERIMENT' in res.environment
+        assert res.group == TestUnit.test_group
+
+    def test_get_parser_2(self):
+        # try to extract all the --foo arguments from the source
+        # and track which ones are flags
+        f = open("../lib/get_parser.py","r")
+        flagargs = set()
+        listargs = set()
+        allargs = []
+        dest = {}
+        for line in f.readlines():
```

Comment:
That's why it checks for each part of the argument flags separately, so if it sees it on a following line it still works. 
The code can be confused by flags that have no double-dash variant, (as we could miss the start) so we have to check for those explicitly.  If there were a lot of non-argparse code in the get_parser.py file, this could also get confused; but its just trying to make a list of all the flags, which ones make lists, and which ones are flags,  and make sure it works to set all of them.   Arguably we should just keep such a list around and just validate it...